t/mojo/promise.t

```diff
@@ -213,6 +213,64 @@ Mojo::IOLoop->one_tick;
 is_deeply \@results, [], 'promises not resolved';
 is_deeply \@errors, ['second'], 'promise rejected';
 
+# All Settled
+$promise  = Mojo::Promise->new->then(sub {@_});
+$promise2 = Mojo::Promise->new->then(sub {@_});
+$promise3 = Mojo::Promise->new->then(sub {@_});
+@results  = ();
+Mojo::Promise->all_settled($promise2, $promise, $promise3)
+  ->then(sub { @results = @_ });
+$promise2->resolve('second');
+$promise3->resolve('third');
+$promise->resolve('first');
+Mojo::IOLoop->one_tick;
+is_deeply \@results,
+  [
+  {status => 'fulfilled', value => ['second']},
+  {status => 'fulfilled', value => ['first']},
+  {status => 'fulfilled', value => ['third']},
+  ],
+  'promises resolved';
+
+# All Settled some rejected
+$promise  = Mojo::Promise->new->then(sub {@_});
+$promise2 = Mojo::Promise->new->then(sub {@_});
+$promise3 = Mojo::Promise->new->then(sub {@_});
+@results  = ();
+Mojo::Promise->all_settled($promise2, $promise, $promise3)
+  ->then(sub { @results = @_ });
+$promise2->reject('second');
+$promise3->resolve('third');
+$promise->resolve('first');
```

Comment:
These tests are not good, they don't actually ensure that the timing was right.

Reply:
And a test with mixed promises is also missing.

Reply:
```
# All Settled some rejected
$promise  = Mojo::Promise->new->timeout(0.1)->resolve('first');
$promise2 = Mojo::Promise->new->timeout(0.2)->reject('second');
$promise3 = Mojo::Promise->new->timeout(0.3)->resolve('third');
@results  = ();
Mojo::Promise->all_settled($promise2, $promise3, $promise)
  ->then(sub { @results = @_ })->wait;
is_deeply(
  \@results,
  [
    {status => 'rejected',  reason => ['second']},
    {status => 'fulfilled', value  => ['third']},
    {status => 'fulfilled', value  => ['first']},
  ],
  'promises resolved'
);
```

What about this test?